alert-database/src/test/java/com/synopsys/integration/alert/database/api/DefaultDescriptorAccessorTest.java

```diff
@@ -0,0 +1,289 @@
+        assertEquals(typeId, registeredDescriptorModel.get().getId());
+        assertEquals(name, registeredDescriptorModel.get().getName());
+        assertEquals(descriptorType, registeredDescriptorModel.get().getType());
+    }
+
+    @Test
+    public void getFieldsForDescriptorTest() throws Exception {
+        final String name = "name-test";
+        final Long typeId = 1L;
+        final ConfigContextEnum configContextEnum = ConfigContextEnum.GLOBAL;
+        final ConfigContextEnum invalidConfigContextEnum = ConfigContextEnum.DISTRIBUTION;
+        final String definedFieldsKey = "defined-field-key-test";
+        Boolean isSensitive = Boolean.TRUE;
+
+        RegisteredDescriptorEntity registeredDescriptorEntity = new RegisteredDescriptorEntity(name, typeId);
+        registeredDescriptorEntity.setId(1L);
+        DescriptorKey descriptorKey = createDescriptorKey("descriptorKey-test");
+        ConfigContextEntity configContextEntity = new ConfigContextEntity(configContextEnum.name());
+        configContextEntity.setId(3L);
+        DefinedFieldEntity definedFieldEntity = new DefinedFieldEntity(definedFieldsKey, isSensitive);
+
+        RegisteredDescriptorRepository registeredDescriptorRepository = Mockito.mock(RegisteredDescriptorRepository.class);
+        DefinedFieldRepository definedFieldRepository = Mockito.mock(DefinedFieldRepository.class);
+        ConfigContextRepository configContextRepository = Mockito.mock(ConfigContextRepository.class);
+        DescriptorTypeRepository descriptorTypeRepository = Mockito.mock(DescriptorTypeRepository.class);
+
+        Mockito.when(registeredDescriptorRepository.findFirstByName(descriptorKey.getUniversalKey())).thenReturn(Optional.of(registeredDescriptorEntity));
+        Mockito.when(configContextRepository.findFirstByContext(configContextEnum.name())).thenReturn(Optional.of(configContextEntity));
+        //Used to test the optional expression
+        Mockito.when(configContextRepository.findFirstByContext(invalidConfigContextEnum.name())).thenReturn(Optional.empty());
+        Mockito.when(configContextRepository.save(Mockito.any())).thenReturn(configContextEntity);
+        Mockito.when(definedFieldRepository.findByDescriptorIdAndContext(Mockito.any(), Mockito.any())).thenReturn(List.of(definedFieldEntity));
+
+        DefaultDescriptorAccessor descriptorAccessor = new DefaultDescriptorAccessor(registeredDescriptorRepository, definedFieldRepository, configContextRepository, descriptorTypeRepository);
+        List<DefinedFieldModel> definedFieldModelList = descriptorAccessor.getFieldsForDescriptor(descriptorKey, configContextEnum);
+        List<DefinedFieldModel> emptyConfigContextDefinedFieldModelList = descriptorAccessor.getFieldsForDescriptor(descriptorKey, invalidConfigContextEnum);
+
+        assertEquals(1, definedFieldModelList.size());
+        assertEquals(definedFieldsKey, definedFieldModelList.get(0).getKey());
```

Comment:
Instead of calling `definedFieldModelList.get(0)` repeatedly, try calling it once and create a local variable and reuse the variable.